### PR TITLE
Refactor PowerShell code from AppVeyor.yml to AppVeyor.psm1

### PR DESCRIPTION
**Pull request by TravisEz13**
Added image requirement to appveyor.yml to make it easier to run on your own AppVeyor instance.
Handle error and retry test results Push

First step of issue #1646 

---

This change is [Reviewable](https://reviewable.io/reviews/powershell/powershell/2121)

### CODE REVIEW

File: tools/appveyor.psm1 [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+function Invoke-AppVeyorTest 
+{
+    #
+    # CoreCLR
+    $env:CoreOutput = Split-Path -Parent (Get-PSOutput -Options (New-PSOptions -Publish -Configuration $buildConfiguration))
+    Write-Host -Foreground Green 'Run CoreCLR tests'
+    $testResultsFile = "$pwd\TestsResults.xml"
+    if(!(Test-Path "$env:CoreOutput\powershell.exe"))
+    {
+        throw "CoreCL PowerShell.exe was not built"
```
Comment (andschwa) [inline tools/appveyor.psm1]: CoreCLR*

File: tools/appveyor.psm1 [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    Update-AppVeyorTestResults -resultsFile $testResultsFileFullCLR
+ 
+
+    #
+    # Fail the build, if tests failed
+    Write-Host -Foreground Green 'Upload CoreCLR test results'
+    $x = [xml](cat -raw $testResultsFile)
+    if ([int]$x.'test-results'.failures -gt 0)
+    {
+    throw "$($x.'test-results'.failures) tests in test/powershell failed"
```
Comment (andschwa) [inline tools/appveyor.psm1]: indentation

File: tools/appveyor.psm1 [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    }
+
+    Write-Host -Foreground Green 'Upload FullCLR test results'
+    $x = [xml](cat -raw $testResultsFileFullCLR)
+
+    if ([int]$x.'test-results'.failures -gt 0)
+    {
+        throw "$($x.'test-results'.failures) tests in test/fullCLR failed"
```
Comment (andschwa) [inline tools/appveyor.psm1]: test/fullclr*

File: tools/appveyor.psm1 [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+# Implements AppVeyor 'on_finish' step
+function Invoke-AppveyorFinish
+{
+      try {
+        # Build packages
+        $packages = Start-PSPackage
+        
+        # Creating project artifact
+        $name = git describe
```
Comment (andschwa) [inline tools/appveyor.psm1]: Does AppVeyor do a full or shallow clone?

Comment (TravisEz13) [inline tools/appveyor.psm1]: full

File: tools/appveyor.psm1 (hunk 1 above)
Comment (TravisEz13) [inline tools/appveyor.psm1]: Resolved

---
In reply to: [76693422](https://github.com/PowerShell/PowerShell/pull/2121#discussion_r76693422) [](ancestors = 76693422)

File: tools/appveyor.psm1 (hunk 2 above)
Comment (TravisEz13) [inline tools/appveyor.psm1]: resolved

---
In reply to: [76693474](https://github.com/PowerShell/PowerShell/pull/2121#discussion_r76693474) [](ancestors = 76693474)

File: tools/appveyor.psm1 (hunk 3 above)
Comment (TravisEz13) [inline tools/appveyor.psm1]: I don't see a difference

---
In reply to: [76693531](https://github.com/PowerShell/PowerShell/pull/2121#discussion_r76693531) [](ancestors = 76693531)

**Pull request merged**